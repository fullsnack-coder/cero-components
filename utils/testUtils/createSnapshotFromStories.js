import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
