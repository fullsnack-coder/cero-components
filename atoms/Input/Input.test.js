import * as stories from './Input.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ atoms / Input ]', () => {
  describe('#Input', () => createSnapshotFromStories('Input', stories))
})
