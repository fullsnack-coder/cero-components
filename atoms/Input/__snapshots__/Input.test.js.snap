// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ atoms / Input ] #Input when 'Input' Default is mounted should render 1`] = `
<DocumentFragment>
  <input
    class="input"
    placeholder="Kingdom slowed"
    type="text"
    value=""
  />
</DocumentFragment>
`;

exports[`[ atoms / Input ] #Input when 'Input' Inline is mounted should render 1`] = `
<DocumentFragment>
  <input
    class="input is-inline"
    placeholder="Kingdom slowed"
    type="text"
    value=""
  />
</DocumentFragment>
`;

exports[`[ atoms / Input ] #Input when 'Input' Types is mounted should render 1`] = `
<DocumentFragment>
  <input
    class="input"
    placeholder="Kingdom slowed"
    type="text"
    value=""
  />
</DocumentFragment>
`;

exports[`[ atoms / Input ] #Input when 'Input' Value is mounted should render 1`] = `
<DocumentFragment>
  <input
    class="input"
    placeholder="Kingdom slowed"
    type="text"
    value="Others mainly cakehole Thorin moldy facial."
  />
</DocumentFragment>
`;
